hub/include/TrayControlWindow.h

```diff
@@ -102,7 +104,11 @@ class TrayControlWindow : public QMainWindow
 public:
   explicit TrayControlWindow(QWidget *parent = 0);
   virtual ~TrayControlWindow();
-
+  static TrayControlWindow* Instance(){
+    static TrayControlWindow *tcw = new TrayControlWindow();
+    return tcw;
+  }
+  void Init (){;} /* do nothing */
```

Comment:
Why this empty method is needed and why you call it later in the code?

Reply:
it uses a constructor to make build connections. So I need it to call my constructor. Actually, I can make something like `(void)TrayControlWindow::Instance()`, but Init functions seems better.

Reply:
Since this is a singleton pattern it will be initialized on first call. Therefore it doesn't needs to be called manually.

Reply:
Yes. But the constructor has a lot of  `connect(CVboxManager::Instance(), &CVboxManager::vm_add, this, &TrayControlWindow::vm_added);` statements. So if I don't call any instances and just use `CVboxManager::Instance().vm_add()` function, then `TrayControlWindow::vm_added` will not be called. Yes, I think it is not very good way to use to use it. I needed just Singleton Instance, but I needed to call the constructor in the beginning, so I just made a new function..

Reply:
`CVboxManager::Instance().vm_add()` will call constructor first.